lib/Onyx.js

```diff
@@ -1043,10 +1062,12 @@ function merge(key, changes) {
                 // After that we merge the batched changes with the existing value
                 let modifiedData = applyMerge(existingValue, [batchedChanges]);
 
-                // For objects, the key for null values needs to be removed from the object to ensure the value will get removed from storage completely.
-                // On native, SQLite will remove top-level keys that are null. To be consistent, we remove them on web too.
-                if (!_.isArray(modifiedData) && _.isObject(modifiedData)) {
-                    modifiedData = _.omit(modifiedData, value => _.isNull(value));
+                modifiedData = cleanObject(modifiedData);
+
+                // On the first write SQLite doesn't use JSON_PATCH, which means, nullish values won't be omitted.
+                // Therefore we remove null values from the `batchedChanges` which are sent to the native implementation, if no existing value is present.
```

Comment:
> which are sent to the native implementation, if no existing value is present.

This doesn't make sense to me. What is "the native implementation"? The native implementation of what? And why does it only get sent  if no existing value is present?